### DISCUSSION TITLE
Modify updater endpoint programmatically

### DISCUSSION THREAD

**Discussion by PhiloNL** [tauri-apps/tauri#4619] [Q&A]
Hi there!

I was wondering if it's possible to modify the update endpoint programmatically?
I want a user-provided value to be passed to the updater endpoint.

```json
  "tauri": {
    "updater": {
      "endpoints": [
          "https://foo.bar/{{current_version}}?key=<user-provided-key>"
      ],
    }
```

Thanks!

**Comment by FabianLars** [comment]
While it's not possible to update the endpoints per se, you can inject [headers](https://docs.rs/tauri/latest/tauri/updater/struct.UpdateBuilder.html#method.header) into the response (on the rust side), so if you can rewrite your server to read the headers instead of the query, you'd be in luck.

If you need to modify the actual url, then we can convert this discussion to a feature request (although i'm not sure if this would be implemented because of security and stuff 🤷)

**Reply by PhiloNL** [reply]
Thanks for the reply @FabianLars, much appreciated! Injecting headers to the request would work as well 😄 
I'm adding Tauri support to Unlock so users can [ship auto-updates](https://twitter.com/Philo01/status/1544710302658433026) protected by license keys.

I could use your help with setting the header, I program mainly in PHP and my Rust knowledge is rather limited 🙈 would you mind sharing a code example on how I can add a header e.g. `Authorization: Bearer 1234` to the request? 🙌

**Reply by PhiloNL** [reply]
I found this [example](https://github.com/tauri-apps/tauri/issues/3896#issuecomment-1107651146). Is that the way to go? If so, would it also be possible to add the header to the default updater without re-implementing the updater logic?
